### PR TITLE
Feature/analyses restart

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

Extensive testing on both the analysis stats and plots jobs have been done to test the restart capabilities of the analyses jobs.  Restart capabilities have been added for both stats and plots jobs.

Also, the ) at the end of the modulefile call is replaced with a }

The $component is added to all the exevs scripts.

The prod_envir module load is removed from the modulefile and is added back to the driver script.  

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

1) For stats restart, you can run any or all jobs using the command:

qsub -v cyc=00 jevs_analyses_urma_grid2obs_stats.sh (you can replace 00 with any number 00 through 23).  

You can either set the wallclock time to half the time that it is listed there (or perhaps just 20 minutes), or just qdel the job after half the stats have been created.  Then qsub the same job again.  You'll see that for small stat files already done, it just bypasses them with a statement that says "RESTART", and then moves on from where it left off.  You should attempt to do this for all three analyses at various times of the day.  

2) For the plots job, you can half the wallclock time to 25-30 minutes or qdel after some plots have been created.  Single png files are now being written to $COMOUTplots.  Thus, when you re-submit the job a second time, the png files in $COMOUTplots are copied to the working directory, so the final tar file can be made at the end.  

Note that before you do this, copy at least one month's worth of analysis files from emc.vpppg.  There are files missing due to issues in the emc.vpppg parallel, but that doesn't really matter for this job - the restart abilities will be tested either way.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @PerryShafran-NOAA  -- I have cloned your fork and and getting underway. Also, go ahead and do a sync fork.

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh:1]: @PerryShafran-NOAA , can these lines be removed?
`export MET_bin_exec=bin`
`export metplus_verbosity=DEBUG`
`export met_verbosity=2`
`export log_met_output_to_metplus=yes`

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA , I have tested the restart for stats. Please review.

**RTMA_RU** ... the job finished even before my reduced runtime. But interestingly, even if the job finishes and you re-run it, the RESTART works. The job runs very quickly b/c it sees the stats files already exist.
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o89671373`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o89671778`

**RTMA** ... I see RESTART in the second log file.  The stats file counts are the same b/t my COMOUT and the emc.vpppg COMOUT. The re-run goes a lot more quickly since many of the stats files are already produced and their processing is skipped.
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_grid2obs_stats.o89671325`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_grid2obs_stats.o89672191`

**URMA** ... I see RESTART in the second log file. The stats file counts are the same b/t my COMOUT and the emc.vpppg COMOUT. The re-run goes a lot more quickly since many of the stats files are already produced and their processing is skipped.
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_urma_grid2obs_stats.o89671492`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_urma_grid2obs_stats.o89672387`

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh
Comment (ShelleyMelchior-NOAA) [reply dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh:1]: They are in a few of the driver scripts.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh:1]: Done

**Comment by PerryShafran-NOAA** [comment]
Looks like everything in stats is working perfectly.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Excellent! I see your updates, too. I will get set up on cactus and test the restart in plots.

**Comment by ShelleyMelchior-NOAA** [comment]
I am testing the plots restart. Please review.

logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o76846856`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o76852530`

**Comment by PerryShafran-NOAA** [comment]
Changed the order of the call to prod_envir and the call to the modulefile, as requested.

Does this need a new re-run?

Perry

**Comment by PerryShafran-NOAA** [comment]
Plot job also is working perfectly as well, including the restart.

Perry

File: dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh [hunk 1]
```diff
@@ -39,7 +39,9 @@ export MODELNAME=rtma
 export modsys=rtma
 export mod_ver=${rtma_ver}
 
-source $HOMEevs/modulefiles/$COMPONENT/${COMPONENT}_${STEP).sh
+module load prod_envir/${prodenvir_ver}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh]: The variable is incorrect. It is ${prod_envir_ver}. Please fix it in all of the drivers.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh]: Corrections made.

**Comment by ShelleyMelchior-NOAA** [comment]
> Does this need a new re-run?

I don't think so. Also, the reason the tests worked w/ the incorrect `$prod_envir_ver` variable is b/c if a version is not specified (effectively what the non-defined `$prodenvir_ver` variable did), it loads the system default.

**Comment by ShelleyMelchior-NOAA** [comment]
I think this PR is square, too. Good?

**Comment by PerryShafran-NOAA** [comment]
I think this one is good to go.

Perry

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
The restart capability for analyses, and the ancillary bits and bobs changes all test as expected. This PR is approved for merge.

**Pull request merged**